### ISSUE TITLE
Lot of bug and don't work

### ISSUE THREAD

**Issue by Bockelbam** [CastagnaIT/plugin.video.netflix#858]
hi,

So here's more than 72 hours spent on this addon, to install a mediacenter on OSMC/Kodi 18.7 Leia on a second TV with a 3b+ rpi.
I've been riding Kodi pretty well for years, so no worries.

So, after having updated a lot of libraries that were not up to date, pycryptodomex that didn't want to install, pyton script errors that bug in all directions, updated kodi 18.5 => 18.7, I finally have an addon that is willing to go for it ...

But main problems now that I can't solve :
- The authentication doesn't work, so ok I bypassed the authenticator.key, but it asks me for the key every time I go down a tree... not very handy if I want to put it in the hands of my teenagers.
- Once connected and arrived on the stream I want, I get an inputstream helper error (Errno 12) cannot allocate memory ... ? the Rpi3 is not powerful enough to run the addon ?
=> I updated the inputstream helper following a similar problem I found on the net: same problem
=> I changed my SD card (i.e. the speed can play) with a new class 3: same thing
=> I changed network connection, wifi to LAN: same thing

now I'm starting to saturate, so if you have an idea for me to finally use this addon I'm a buyer.

**Comment by jakermx** [comment]
Have tried Libreelec or similars, it took me less than 30 minutes to get it worked from the scratch on my RPI 3 B+ and RPI2B

**Comment by Bockelbam** [comment]
It doesn't say anywhere that you have to use LibreElec exclusively, and I prefer to use OSMC

**Comment by jakermx** [comment]
OK, can you share your System Information , the detailed procedure you are using to install it, and debug logs  so we can try to support you. Please

**Comment by CastagnaIT** [comment]
> (Errno 12) cannot allocate memory

this error means that your device not have memory to manage the operations
it is strange that come from inputstream helper addon
but you have not provide any log so is impossible understand what exactly happen...

> The authentication doesn't work, so ok I bypassed the authenticator.key, but it asks me for the key every time I go down a tree... not very handy if I want to put it in the hands of my teenagers.

also this is strange,
i use the key from more that a week, inserting it only the first time (no more ask to login)
it is possible that you have used the same key with more devices or it is related to other problems

**Comment by Bockelbam** [comment]
Thx for your respond

here is my last log, it's at the very end
[kodi.old.log](https://github.com/CastagnaIT/plugin.video.netflix/files/5208642/kodi.old.log)

**Comment by jakermx** [comment]
NF Background Service is not started, so all calls to any NF related fails, is it Enabled, can you navigate thru profiles  ?

**Comment by CastagnaIT** [comment]
are you sure that your RAM have enough space to run all Kodi + addons?
these errors persist:

> OSError: [Errno 12] Cannot allocate memory

i do not know RPI, but it is possible that you do not have a swap partition to ensure memory peaks?

**Comment by Bockelbam** [comment]
> NF Background Service is not started, so all calls to any NF related fails, is it Enabled, can you navigate thru profiles ?

yes, here it's just because I threw it too soon ...
Otherwise I wouldn't have been able to get into it and have my authentication problems :)

**Comment by Bockelbam** [comment]
> are you sure that your RAM have enough space to run all Kodi + addons?
> these errors persist:
> 
> > OSError: [Errno 12] Cannot allocate memory
> 
> i do not know RPI, but it is possible that you do not have a swap partition to ensure memory peaks?

I use the RPI in its original configuration and without swap partition, I use all Kodi without any problem with all these addons without memory saturation (never more than 80%), so far, so maybe for Netflix it's not enough, because basic OSMC doesn't create a swap partition, and that's why "maybe" LireElec creates a basic one ... I don't know right away.
But it may indeed be a lead

**Comment by jakermx** [comment]
I have a RPI3B+ as spare, let me download OSMC and try it.... but I have to go work now, I will do it later today and will keep you posted

**Comment by Bockelbam** [comment]
then it works!
I modified the rpi config, because I realized that the 1Gb of RAM was not allocated, but only 750Mb.
So I modified it to allocate more memory, and I took the opportunity to overcloak the RPI slightly

And as luck would have it, it no longer requires me to authenticate every tree structure in NF, and I was able to go as far as reading the stream.

To make sure it was indeed this, I reset the memory values as before, and I reproduced the bugs

thank you for your assistance, it allowed me to give me leads of investigations
